src/Microsoft.CodeQuality.Analyzers.Exp/Core/FlowAnalysis/Analysis/StringContentAnalysis/StringContentAnalysis.StringContentDataFlowOperationWalker.cs

```diff
@@ -0,0 +1,154 @@
+                        var leftValue = Visit(operation.LeftOperand, argument);
+                        var rightValue = Visit(operation.RightOperand, argument);
+                        return leftValue.MergeBinaryAdd(rightValue, operation);
+
+                    default:
+                        return base.VisitBinaryOperator(operation, argument);
+                }
+            }
+
+            public override StringContentAbstractValue VisitCompoundAssignment(ICompoundAssignmentOperation operation, object argument)
+            {
+                StringContentAbstractValue value;
+                switch (operation.OperatorKind)
+                {
+                    case BinaryOperatorKind.Add:
+                    case BinaryOperatorKind.Concatenate:
+                        var leftValue = Visit(operation.Target, argument);
+                        var rightValue = Visit(operation.Value, argument);
+                        value = leftValue.MergeBinaryAdd(rightValue, operation);
+                        break;
+
+                    default:
+                        value = base.VisitCompoundAssignment(operation, argument);
+                        break;
+                }
+
+                SetAbstractValueForAssignment(operation.Target, value);
+                return value;
+            }
+
+            public override StringContentAbstractValue VisitNameOf(INameOfOperation operation, object argument)
+            {
+                var nameofValue = base.VisitNameOf(operation, argument);
+                if (operation.ConstantValue.HasValue && operation.ConstantValue.Value is string value)
+                {
+                    return new StringContentAbstractValue(literal: value);
+                }
+
+                return nameofValue;
+            }
+
+            public override StringContentAbstractValue VisitObjectCreation(IObjectCreationOperation operation, object argument)
+            {
+                // TODO: Analyze string constructor
+                return base.VisitObjectCreation(operation, argument);
+            }
+
+            public override StringContentAbstractValue VisitFieldReference(IFieldReferenceOperation operation, object argument)
+            {
+                var value = base.VisitFieldReference(operation, argument);
+
+                // Handle "string.Empty"
+                if (operation.Field.Name.Equals("Empty", StringComparison.Ordinal) &&
+                    operation.Field.ContainingType.SpecialType == SpecialType.System_String)
+                {
+                    return new StringContentAbstractValue(literal: string.Empty);
+                }
+
+                return value;
+            }
+
+            public override StringContentAbstractValue VisitInvocation_NonLambdaOrDelegateOrLocalFunction(IInvocationOperation operation, object argument)
+            {
+                // TODO: Handle invocations of string methods (Format, SubString, Replace, etc.)
```

Comment:
>TODO [](start = 19, length = 4)

Work item?

Reply:
`Concat` is another method to look out for.